ui/panels-plugin/src/plugins/time-series-chart/utils/palette-gen.ts

```diff
@@ -23,3 +27,26 @@ export function getRandomColor(identifier: string): string {
   const color = `hsla(${~~(180 * hash)},50%,50%,0.8)`;
   return color;
 }
+
+/**
+ * Get line color as well as color for tooltip and legend, account for whether palette is 'Cateogrical' or 'Auto' (generative)
+ */
+export function getSeriesColor(
+  name: string,
+  seriesCount: number,
+  palette: string[],
```

Comment:
Actually, here would be the fun place to fallback to a default palette of colors!

Reply:
I'll add a comment describing more about the ECharts theme! The palette being passed in here is our default palette from `useChartsTheme` [here](https://github.com/perses/perses/blob/6a9432648cb642d48e4c7086a40b2d0bcfafdbf3/ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx#L59). Internally we have a different palette defined in our ECHARTS_THEME_OVERRIDES which lets us use different colors for Chronosphere vs. Perses (RedHat also wants to leverage this to use their design system colors)

The FALLBACK_PALETTE_COLOR above isn't really necessary since the ECharts theme palette will always be set, planning to play around with the types to see if I can remove it

Reply:
@jgbernalp maybe you can be interested by that :)